### ISSUE TITLE
AWS::StepFunctions::StateMachine - false-positive finding `E3601` for `Assign` in `Choice` state

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4049]
### CloudFormation Lint Version

cfn-lint 1.32.0

### What operating system are you using?

Windows 11

### Describe the bug

It is possible to assign Variables in Step Functions using JSONPath.
In `Choice` types this is also possible (It works to deploy this CloudFormation template):

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Choice
        States:
          Choice:
            Type: Choice
            Choices:
              - Next: Fail
                Variable: $
                IsPresent: true
            Default: Fail
            Assign: # <----- Creates a Variable for the Default choice
              Key: Value
          Fail:
            Type: Fail
```

### Expected behavior

cfn-lint doesn't give `E3601` when using `Assign` for a `Default` choice.

### Reproduction template

See above.
